text/005-co-location.md

```diff
@@ -0,0 +1,1077 @@
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+## Before RFC Merge
+
+1. **Should we support runtime variant registration?**
+
+   - **Question**: Can developers add new variants after build time?
+   - **Consideration**: Would enable dynamic theming but requires runtime CSS injection
+   - **Trade-off**: Performance vs flexibility
+
+2. **How to handle CSS specificity conflicts?**
+
+   - **Question**: What if compound variants conflict with base styles?
+   - **Current behavior**: Later classes in className string win (CSS cascade)
+   - **Alternative**: Use `!important` for compound variants?
+
+3. **Should we support nested variants?**
+   ```tsx
+   variants: {
+   	size: {
+   		small: {
+   			variants: {
+   				compact: {
+   					padding: '2px';
+   				}
+   			}
+   		}
+   	}
+   }
+   ```
+   - **Question**: Is nesting needed or does it add unnecessary complexity?
+
+## Before Stabilization
+
+1. **Performance benchmarks**
+
+   - Measure runtime overhead vs styled-components, emotion
+   - Bundle size comparison for various use cases
+   - Type-checking performance with large component libraries
+
+2. **Error messages**
+
+   - Improve error messages for Babel transformation failures
+   - Better TypeScript errors for invalid variant combinations
+
+3. **Developer tools**
+   - Browser extension for inspecting variant state
+   - Debug mode showing which variants are applied
+
+## Out of Scope (Future RFCs)
+
+1. **CSS extraction optimization**
+
+   - Critical CSS inlining
+   - Automatic CSS splitting by route
+
+2. **Design system integration**
+
+   - Figma plugin for generating variants from designs
+   - Theme tokens integration
+
+3. **Animation support**
+   - Transition between variant states
+   - Keyframe animations with variants
+
+# Future possibilities
+
+[future-possibilities]: #future-possibilities
+
+## 1. Variant Inheritance
+
+Allow variants to inherit from other variants:
+
+```tsx
+const Button = styled('button', {
+	variants: {
+		color: {
+			primary: { backgroundColor: 'blue' },
+			primaryHover: {
+				extends: 'primary',
+				backgroundColor: 'darkblue',
+			},
+		},
+	},
+});
+```
+
+## 2. Responsive Variants
+
+Media-query-aware variants:
+
+```tsx
+const Text = styled('span', {
+	variants: {
+		size: {
+			responsive: {
+				'@initial': { fontSize: '12px' },
+				'@md': { fontSize: '16px' },
+				'@lg': { fontSize: '20px' },
+			},
+		},
+	},
+});
+
+<Text size="responsive">Responsive Text</Text>;
+```
+
+## 3. Variant Slots (Multi-Part Components)
+
+Support for components with multiple styled parts:
+
+```tsx
+const Card = styled('div', {
+	slots: {
+		root: { padding: '20px' },
+		header: { fontWeight: 'bold' },
+		body: { marginTop: '10px' },
+	},
+	variants: {
+		elevated: {
+			true: {
+				root: { boxShadow: '0 2px 4px rgba(0,0,0,0.1)' },
+			},
+		},
+	},
+});
+
+<Card elevated>
+	<Card.Header>Title</Card.Header>
+	<Card.Body>Content</Card.Body>
+</Card>;
+```
```

Comment:
It seems like slot recipe can be replaced with an instance of [`rules.multiple()`](https://github.com/mincho-js/working-group/blob/main/text/004-methods.md#3-multiple).